tests/mock_gpu_test.py

```diff
@@ -27,7 +27,7 @@
 NUM_SHARDS = 16
 
 
-@jtu.with_config(use_mock_gpu_client=True, mock_num_gpus=NUM_SHARDS)
+@jtu.with_config(mock_num_processes=NUM_SHARDS)
 class MockGPUTest(jtu.JaxTestCase):
 
   def testMockWithSharding(self):
```

Comment:
The `testMockWithSharding` test is somewhat misleading because on 8-GPU machines it actually mocks 128 GPUs, but then creates a global mesh for only 16 of them. Do you think it would make sense to make the test less misleading (perhaps in a different PR), say by creating a mesh with NUM_SHARDS*jax.local_device_count() devices?

In addition to that, the test does not seem to protect against broken mocking - `jtu.create_global_mesh` simply skips the test if there is not enough devices, so removing the mock_num_processes setting does not cause the to fail.

Reply:
I have updated the test to use all  devices and to check that MLIR is sharded to all the mocked GPUs.

In some ways, this is an elaborate version of the one-line test (`self.assertEqual(jax.device_count(), jax.local_device_count() * NUM_SHARDS)`).